soups/show-and-tell/show-and-tell-37.snip.markdown

```diff
@@ -0,0 +1,120 @@
+
+[spike]: https://en.wikipedia.org/wiki/Spike_(software_development\)
+
+## Insights from your test suite
+
+Newcomer to Show and Tell [Chris Zetter](chris-zetter) showed us a side-project he's been working on which aims to extract insights from the historical runs of a projects test suite. Continuous Integration systems such as Jenkins allow you to see how long each build has taken but Chris was interested in more granular data - which individual tests have become slower, how many new tests have been added and which are the most likely to fail, for example.
+
+Users of the system install add a Ruby gem to the project's Gemfile (Chris has this working with RSpec and Minitest so far) which sends data about the test suite to a central server. A web application then tabulates and graphs this data to allow you to understand it.
+
+![Chris demos his application](/images/blog/zetter-demo.jpg)
+
+Chris talked a bit about what he plans to do next - including trialling the service with some real-world applications. We also had a discussion about one of Chris's key challenges - turning the data he is collecting into actionable insights. He mentioned that seeing those tests that occasionally fail has already been useful - it's a strong indicator of an intermittent test that could be improve or replaced. The working title of the project is URF - short for Useful, Fast, Reliable. Chris said he feels the data he already collects helps to identify slow and unreliable tests - but there's a lot of scope for collecting information that tells the user whether an individual test is useful.
+
+There was a lot of support for Chris's project in the room - with several of us saying that we'd find such a service useful for the applications we work on. It'll be fascinating to see where Chris takes this next.
+
+## Things found on old laptops
+
+Before [Ben](ben-griffiths) sold an old laptop he thought he'd better have a look on the disk to see if there was anything valuable. He found a few old projects that he'd worked on while CTO at [MakieLab](http://mymakie.com/).
+
+These projects were part of the system which allowed users of the website to design, and then have 3D-printed, their own customised doll.
+
+![An early Makie prototype doll](/images/blog/makie-doll.jpg)
+
+The doll's face was a composite of more than 30 different models which each varied one parameter such as the size of the nose, the length of the ears, or the spacing between the eyes. By moving sliders on the web site users could blend between these different models to vary the face.
+
+At the time Ben built the system, the support for Adobe Flash was starting to wane. Ben used [WebGL]() to render the model in supported browsers and also developed a pure-JavaScript rendering engine as a fallback. Ben had unearthed one of the first versions of the code where he had succeeded in rendering a single eye ball.
```

Comment:
Does this `[WebGL]()` link work? I thought you needed square brackets instead of parentheses.

Reply:
Fixed in ed56e28f62f51a57f73ccf49c48ba2bb502e81fa.